File: conversation-websocket-tools/python/create_tool.py
from phonic import Phonic
import os
from dotenv import load_dotenv

load_dotenv(".env.local")

client = Phonic(
    api_key=os.getenv("PHONIC_API_KEY"),
)

client.tools.create(
    name="find_flights_sync",
    description="Find one-way flights on a given date from an origin to a destination airport.",
    type="custom_websocket",
    execution_mode="sync",
    parameters=[
        {
            "type": "string",
            "name": "date",
            "description": "The date of the flight in YYYY-MM-DD format",
            "is_required": True,
        },
        {
            "type": "string",
            "name": "from_airport",
            "description": "The origin airport code",
            "is_required": True,
        },
        {
            "type": "string",
            "name": "to_airport",
            "description": "The destination airport code",
            "is_required": True,
        },
    ],
)

File: node/conversation-websocket-tools/create-tool.ts
import { PhonicClient } from "phonic";
import { phonicApiKey } from "./env-vars";

const client = new PhonicClient({
  apiKey: phonicApiKey,
});

async function createTool() {
  await client.tools.create({
    name: "get_temperature",
    description:
      "Get the current temperature and weather conditions for a specific city.",
    type: "custom_websocket",
    execution_mode: "sync",
    parameters: [
      {
        type: "string",
        name: "city",
        description: "The name of the city to get the temperature for",
        is_required: true,
      },
    ],
  });
}

createTool();
